test/unit/math/serializer.hpp

```diff
@@ -60,7 +60,8 @@ struct deserializer {
    * @param x pattern argument to determine result shape and size
    * @return deserialized value with shape and size matching argument
    */
-  template <typename U>
+  template <typename U, require_stan_scalar_t<U>* = nullptr,
+            require_not_complex_t<U>* = nullptr>
   T read(const U& x) {
     return vals_[position_++];
   }
```

Comment:
I don't see any overload for complex scalars. Is it intentionally missing?

Reply:
That's because there's already a complex scalar overload on L80, but because I added complex types to ```is_stan_scalar```, this templating became ambiguous